scripts/kind-e2e/lib_operator_verify_subm.sh

```diff
@@ -1,491 +0,0 @@
-
-  validate_equals '.metadata.namespace' $subm_ns
-  validate_equals '.apiVersion' 'submariner.io/v1alpha1'
-  validate_equals '.kind' 'Submariner'
-  validate_equals '.metadata.name' $deployment_name
-  validate_equals '.spec.brokerK8sApiServer' $SUBMARINER_BROKER_URL
-  # TODO: every cluster must have it's own token / SA (not working when using bundle/acm)
-  # validate_not_equals '.spec.brokerK8sApiServerToken' $SUBMARINER_BROKER_TOKEN
-  validate_equals '.spec.brokerK8sCA' $SUBMARINER_BROKER_CA
-  validate_equals '.spec.brokerK8sRemoteNamespace' $SUBMARINER_BROKER_NS
-  validate_equals '.spec.ceIPSecDebug' $ce_ipsec_debug
-  validate_equals '.spec.ceIPSecNATTPort' $ce_ipsec_nattport
-  validate_equals '.spec.repository' $subm_gateway_image_repo
-  validate_equals '.spec.version' $subm_gateway_image_tag
-  validate_equals '.spec.broker' $subm_broker
-  echo "Generated cluster id: $(jq -r '.spec.clusterID' $json_file)"
-  validate_equals '.spec.debug' $subm_debug
-  validate_equals '.spec.namespace' $subm_ns
-  validate_equals '.spec.natEnabled' $natEnabled
-
-}
-
-function verify_subm_op_pod() {
-  subm_operator_pod_name=$(kubectl get pods --namespace=$subm_ns -l name=$operator_deployment_name -o=jsonpath='{.items..metadata.name}')
-  if [[ -z "${subm_operator_pod_name}" ]]; then
-    subm_operator_pod_name=$(kubectl get pods --namespace=$subm_ns -l control-plane=$operator_deployment_name -o=jsonpath='{.items..metadata.name}')
-  fi
-
-  # Show SubM Operator pod info
-  kubectl get pod $subm_operator_pod_name --namespace=$subm_ns -o json
-
-  # Verify SubM Operator pod status
-  kubectl get pod $subm_operator_pod_name --namespace=$subm_ns -o jsonpath='{.status.phase}' | grep Running
-
-  # Show SubM Operator pod logs
-  kubectl logs $subm_operator_pod_name --namespace=$subm_ns
-
-  # TODO: Verify logs?
-}
-
-function validate_pod_container_equals() {
-  validate_equals ".spec.containers[].${1}" $2
-}
-
-function validate_pod_container_has() {
-  local json_filter=".spec.containers[].${1}"
-  local expected=$2
-  [[ $(jq -r -M $json_filter $json_file) =~ $expected ]]
-}
-
-function validate_pod_container_env() {
-  local var_name=$1
-  local expected=$2
-  [[ $(jq -r -M ".spec.containers[].env[] | select(.name==\"${var_name}\").value" $json_file) = $expected ]]
-}
-
-function verify_subm_gateway_pod() {
-  kubectl wait --for=condition=Ready pods -l app=$gateway_deployment_name --timeout=120s --namespace=$subm_ns
-
-  subm_gateway_pod_name=$(kubectl get pods --namespace=$subm_ns -l app=$gateway_deployment_name -o=jsonpath='{.items..metadata.name}')
-
-  json_file=/tmp/${subm_gateway_pod_name}.${cluster}.json
-  kubectl get pod $subm_gateway_pod_name --namespace=$subm_ns -o json | tee $json_file
-
-  validate_pod_container_equals 'image' "${subm_gateway_image_repo}/submariner-gateway:local"
-  validate_pod_container_has 'securityContext.capabilities.add' 'net_admin'
```

Comment:
Am I right in understanding that the new tests don't actually assert details at this level? The just check for the presence of broadly-scoped K8s objects?

Reply:
Yes as this knowledge is internal to operator and tested there, here its not necessary nor helpful to have these white box tests as the operator takes care of these details.